graphql_compiler/schema_transformation/rename_query.py

```diff
@@ -131,18 +169,56 @@ def leave_selection_set(
     def enter_field(
         self, node: FieldNode, key: Any, parent: Any, path: List[Any], ancestors: List[Any]
     ) -> Union[FieldNode, VisitorAction]:
-        """Rename root vertex fields."""
+        """Rename fields."""
         # For a Field to be a root vertex field, it needs to be the first level of
         # selections (fields in more nested selections are normal fields that should not be
         # modified)
         # As FragmentDefinition is not allowed, the parent of the selection must be a query
         # As a query may not start with an inline fragment, all first level selections are
         # fields
         if self.selection_set_level == 1:
-            renamed_node = self._rename_name(node)
-            if renamed_node is node:  # Name unchanged, continue traversal
-                return None
-            else:  # Name changed, return new node, `visit` will make shallow copies along path
-                return renamed_node
+            self.current_type_name.append(node.name.value)
+        else:
+            # Entered a regular field and we want to find its type
+            current_type_name = self.current_type_name[-1]
+            current_type = self.schema.get_type(current_type_name)
+            if not current_type:
```

Comment:
Do we expect this to be `None` if this branch gets hit? If so, I have a mild preference for the more explicit `is None` check because otherwise we have to know what could make a type falsy. `if not current_type` does `bool(current_type)` under the hood, which usually does the right thing, but if someone defines a `__bool__` method then anything goes. With `is None`, I know exactly what the code does with no wiggle room.

I realize this is the opposite of the Google style guide, so if you decide you want to keep it this way, that's fine too.
```suggestion
            if current_type is None:
```